File: libdata/src/main/java/com/simprints/libdata/models/realm/RealmConfig.java
package com.simprints.libdata.models.realm;

import android.support.annotation.NonNull;

import java.util.Arrays;

import io.realm.Realm;
import io.realm.RealmConfiguration;

public class RealmConfig {
    public static RealmConfiguration get(@NonNull String localDbKey) {
        String dbName = String.format("%s.realm", localDbKey);

        byte[] key = Arrays.copyOf(localDbKey.getBytes(), 64);

        return new RealmConfiguration.Builder()
                .name(dbName)
                .schemaVersion(1)
                .migration(new Migration())
                .encryptionKey(key)
                .modules(Realm.getDefaultModule())
                .build();
    }
}

File: id/src/main/java/com/simprints/id/data/db/local/realm/RealmConfig.kt
package com.simprints.id.data.db.local.realm

import com.simprints.id.data.db.local.realm.Migration.Companion.REALM_SCHEMA_VERSION
import io.realm.Realm
import io.realm.RealmConfiguration

object RealmConfig {

    fun get(databaseName: String, key: ByteArray): RealmConfiguration = RealmConfiguration
        .Builder()
        .name("$databaseName.realm")
        .schemaVersion(REALM_SCHEMA_VERSION)
        .migration(Migration())
        .encryptionKey(key)
        .addModulesIfNotNull()
        .build()

    private fun RealmConfiguration.Builder.addModulesIfNotNull(): RealmConfiguration.Builder =
        Realm.getDefaultModule()?.let { this.modules(it) }?: this
}
